src/argprocessing.cpp

```diff
@@ -928,6 +928,11 @@ process_arg(Context& ctx,
 
   // Rewrite to relative to increase hit rate.
   args_info.input_file = Util::make_relative_path(ctx, args[i]);
+  if (i < ctx.orig_args.size()) {
+    // Rewrite to relative in original args in case hash_full_command_line
+    // is set (-frecord-gcc-switches) - to increase hit rate
+    ctx.orig_args.replace(i, args_info.input_file);
```

Comment:
Sorry, but `process_arg` is not allowed to modify `ctx.orig_args` like that. One reason is that `ctx.orig_args` needs to stay exactly the same so that ccache can execute the original command line on failure.

Also, won't we have the same problem if there are other absolute paths than the input file on the command line, for instance `-I/home/user/test1/include`? I'm thinking there should be a more generic solution to this.

A simple and generic fix could be to add a [sloppiness option](https://ccache.dev/manual/4.4.1.html#config_sloppiness) for not hashing the full command line when `-frecord-gcc-switches`, as mentioned in https://github.com/ccache/ccache/issues/804#issuecomment-784493246. What do you think about that solution?

Reply:
Thanks for the explanation on `ctx.orig_args` usage to why it's not appropriate to modify it.
> Also, won't we have the same problem if there are other absolute paths than the input file on the command line, for instance -I/home/user/test1/include? I'm thinking there should be a more generic solution to this.
Agreed, didn't check this.
> A simple and generic fix could be to add a sloppiness option for not hashing the full command line when -frecord-gcc-switches, as mentioned in #804 (comment). What do you think about that solution?
Sloppiness option looks more clean and will resolve both input and include absolute paths being cached. Will put a new merge request with sloppiness option instead of this one.